Source/Core/DolphinQt/GameList/GameListModel.cpp

```diff
@@ -257,8 +257,12 @@ bool GameListModel::ShouldDisplayGameListItem(int index) const
 {
   const UICommon::GameFile& game = *m_games[index];
 
+  const bool show_filename = Config::Get(Config::MAIN_GAMELIST_COLUMN_FILE_NAME);
   if (!m_term.isEmpty() &&
-      !QString::fromStdString(game.GetName(m_title_database)).contains(m_term, Qt::CaseInsensitive))
+      !QString::fromStdString(game.GetName(m_title_database))
+           .contains(m_term, Qt::CaseInsensitive) &&
+      !(QString::fromStdString(game.GetFileName()).contains(m_term, Qt::CaseInsensitive) &&
+        show_filename))
```

Comment:
`show_filename && string.contains(m_term)` would be better due to [short-circuit evaluation](https://en.wikipedia.org/wiki/Short-circuit_evaluation); checking whether one string contains another string is a slower operation that it would be better to skip if the filename column is hidden. In this case it probably doesn't matter that much (even if someone has *all* of the GameCube and Wii discs, that's only about 5000 of them, which probably could still be handled in milliseconds), but it's something to keep in mind for other situations.

This logic is also getting a bit messy, though; maybe something like this would be clearer? (I haven't tested this code and am not sure if it actually compiles/is formatted correctly)

```C++
if (!m_term.isEmpty())
{
  const bool matches_title = !QString::fromStdString(game.GetName(m_title_database)).contains(m_term, Qt::CaseInsensitive));
  const bool filename_visible = Config::Get(Config::MAIN_GAMELIST_COLUMN_FILE_NAME);
  const bool matches_filename = filename_visible && !(QString::fromStdString(game.GetFileName()).contains(m_term, Qt::CaseInsensitive);
  if (!(matches_title || matches_filename))
  {
    return false;
  }
}
```